shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -1002,5 +1002,56 @@ TEST_F(FlutterWindowsEngineTest, InnerWindowHidden) {
             AppLifecycleState::kInactive);
 }
 
+TEST_F(FlutterWindowsEngineTest, EnableLifecycleState) {
```

Comment:
Could we have a test for this scenario:

1. The Dart app sends the initialized system message
2. The C++ test updates a window state
3. The test verifies the Dart app receives the app lifecycle update